letsencrypt/configuration.py

```diff
@@ -82,6 +84,21 @@ def http01_port(self):  # pylint: disable=missing-docstring
         else:
             return challenges.HTTP01Response.PORT
 
+    def determine_user_agent(self, authenticator, installer):
```

Comment:
I wonder if this is the best place for this function. In general, this class is just a wrapper around `argparse.Namespace`.

What do you think of the approach of modifying this function so it returns the UA and put it in `client.py`. When `letsencrypt.client.Client` is initiated, it has the `authenticator` and `installer`. `__init__` can call this function and pass it into `_acme_from_config_key` which should take the UA as an optional argument. Thoughts?